cmd/release/main.go

```diff
@@ -163,6 +175,237 @@ func main() {
+						if matches == nil {
+							return fmt.Errorf("stable-date must be of form %s", releaseDateStringPattern)
+						}
+					}
+
+					repoFullName := c.String("repo")
+					repoRegexp := regexp.MustCompile(`^([^/]+)/([^/]+)$`)
+					matches := repoRegexp.FindStringSubmatch(repoFullName)
+					if matches == nil {
+						return fmt.Errorf("invalid repository name format. Must be 'owner/repo'")
+					}
+					repoOwner := matches[1]
+					repoName := matches[2]
+
+					// Prepare template data
+					data := make(map[string]interface{})
+					data["CreateOnGitHub"] = createOnGitHub
+					data["Type"] = releaseType
+					data["Tag"] = releaseVersion.String()
+					data["NextTag"] = releaseVersion.IncPatch().String()
+					data["Level"] = releaseLevel
+					data["NetworkUpgrade"] = networkUpgrade
+					data["NetworkUpgradeDiscussionLink"] = discussionLink
+					data["NetworkUpgradeChangelogEntryLink"] = changelogLink
+					data["RC1DateString"] = rc1Date
+					data["StableDateString"] = stableDate
+
+					// Render the issue template
+					issueTemplate, err := os.ReadFile("documentation/misc/RELEASE_ISSUE_TEMPLATE.md")
+					if err != nil {
+						return fmt.Errorf("failed to read issue template: %w", err)
+					}
+					// Sprig used for String contains and Lists
+					tmpl, err := template.New("issue").Funcs(sprig.FuncMap()).Parse(string(issueTemplate))
+					if err != nil {
+						return fmt.Errorf("failed to parse issue template: %w", err)
+					}
+					var issueBodyBuffer bytes.Buffer
+					err = tmpl.Execute(&issueBodyBuffer, data)
+					if err != nil {
+						return fmt.Errorf("failed to execute issue template: %w", err)
+					}
+
+					// Prepare issue creation options
+					issueTitle := fmt.Sprintf("Lotus %s v%s Release", releaseType, releaseTag)
+					issueBody := issueBodyBuffer.String()
+
+					// Remove duplicate newlines before headers and list items since the templating leaves a lot extra newlines around.
+					// Extra newlines are present because go formatting control statements done within HTML comments rather than using {{- -}}.
+					// HTML comments are used instead so that the template file parses as clean markdown on its own.
+					// In addition, HTML comments were also required within "ranges" in the template.
+					// Using HTML comments everywhere keeps things consistent.
+					re := regexp.MustCompile(`\n\n+([^#*\[\|])`)
+					issueBody = re.ReplaceAllString(issueBody, "\n$1")
+
+					if !createOnGitHub {
+						// Create the URL-encoded parameters
+						params := url.Values{}
+						params.Add("title", issueTitle)
+						params.Add("body", issueBody)
+						params.Add("labels", "tpm")
+
+						// Construct the URL
+						issueURL := fmt.Sprintf("https://github.com/%s/issues/new?%s", repoFullName, params.Encode())
+
+						debugFormat := `
+Issue Details:
+=============
+Title: %s
+
+Body:
+-----
+%s
+
+URL to create issue:
+-------------------
+%s
+`
+						fmt.Printf(debugFormat, issueTitle, issueBody, issueURL)
```

Comment:
Typically in urfave/cli you should `_, _ = fmt.Fprintf(c.App.Writer, "tmpl", arg1, arg2)`, and you also have access to `c.App.ErrWriter`. They point to stdout and stderr by default but it makes the app code more flexible.

Reply:
Thanks.  I updated this.